openpype/hosts/nuke/api/lib.py

```diff
@@ -2094,9 +2101,29 @@ def set_root_colorspace(self, imageio_host):
 
         # set ocio config path
         if config_data:
-            current_ocio_path = os.getenv("OCIO")
-            if current_ocio_path != config_data["path"]:
-                message = """
+            log.info("OCIO config path found: `{}`".format(
+                config_data["path"]))
+
+            # check if there's a mismatch between environment and settings
+            wrong_environment = self._is_settings_different_from_environment(
+                config_data)
+
+            # if there's no mismatch between environment and settings
+            if not wrong_environment:
+                self._set_ocio_config_path_to_workfile(config_data)
```

Comment:
The logic here to me would be much clearer if it would just read:

```suggestion
            config_path = config_data["path"]
            log.info("OCIO config path found: `{}`".format(config_path))
            
            if config_path == os.getenv("OCIO"):
                self._set_ocio_config_path_to_workfile(config_data)
```

The double negative check of "is it different" but then `not different` seems a bit confusing to read.

Also, I suspect the OCIO path checks might need to also check with normalised `os.path` unless we know for certain the env var is set exactly (matching case + forward/backward slashes) to the value of `config_data["path"]`.

Reply:
Thanks for the suggestion. I'll process it differently, but yeah, I agree the double negative is confusing.